src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -0,0 +1,83 @@
+package frc.robot.subsystems;
+
+import com.spikes2212.command.DashboardedSubsystem;
+import com.spikes2212.util.Limelight;
+import edu.wpi.first.cameraserver.CameraServer;
+import edu.wpi.first.math.geometry.Pose3d;
+import org.photonvision.PhotonCamera;
+import org.photonvision.targeting.PhotonPipelineResult;
+
+public class Vision extends DashboardedSubsystem {
+
+    public static final String PHOTON_VISION_CAMERA_NAME = "photonvision";
+
+    public static final int PIPELINE_CONE_INDEX = 0;
+    public static final int PIPELINE_CUBE_INDEX = 1;
+
+    public static final int PIPELINE_HIGH_RRT_INDEX = 0;
+    public static final int PIPELINE_LOW_RRT_INDEX = 1;
+    public static final int PIPELINE_APRILTAG_INDEX = 2;
+
+    private static Vision instance;
+
+    private final PhotonCamera photonCamera;
+    private final Limelight limelight;
+
+    public static Vision getInstance() {
+        if (instance == null) {
+            instance = new Vision("vision", new PhotonCamera(PHOTON_VISION_CAMERA_NAME), new Limelight());
+        }
+        return instance;
+    }
+
+    private Vision(String namespaceName, PhotonCamera photonCamera, Limelight limelight) {
+        super(namespaceName);
+        this.photonCamera = photonCamera;
+        this.limelight = limelight;
+        configureDashboard();
+    }
+
+    public double getPhotonVisionYaw() {
+        PhotonPipelineResult result = photonCamera.getLatestResult();
+        if (result.hasTargets()) {
+            return result.getBestTarget().getYaw();
+        }
+        return 0; // if no target is detected
```

Comment:
Problematic if you want to center on a cube/cone when no target is detected. Either change this to an arbitrary constant that will function well with the PID, or change the command to handle a situation where there is no target.
(I recomment doing this in the command because SOLID)

Reply:
The "handling a situation" is simply returning 0. If there is no target, you don't want to immediately cancel the command (what if the camera just didn't find the object for half a second? you would want to continue still when it finds it) so it just ends after the wait time in the PID. Also, as long that there is no target - robot doesn't move